File: apps/api/src/application/support/activity-logger.ts
import type { ActivityLogRepository } from '../../infrastructure/repositories/activity-log-repository.js';

export class ActivityLogger {
	constructor(private readonly logs: ActivityLogRepository) {}

	log(input: {
		adminUserId: number | null;
		action: string;
		entityType: string;
		entityId?: string | null;
		details?: Record<string, unknown> | null;
	}): Promise<void> {
		return this.logs.insert(input);
	}
}

File: apps/python-api/src/yourmedportal_api/application/support/activity_logger.py
from __future__ import annotations

from typing import Any

from yourmedportal_api.infrastructure.repositories.activity_log_repository import ActivityLogRepository


class ActivityLogger:
    def __init__(self, logs: ActivityLogRepository) -> None:
        self._logs = logs

    def log(
        self,
        *,
        admin_user_id: int | None,
        action: str,
        entity_type: str,
        entity_id: str | None = None,
        details: dict[str, Any] | None = None,
    ) -> None:
        self._logs.insert(
            admin_user_id=admin_user_id,
            action=action,
            entity_type=entity_type,
            entity_id=entity_id,
            details=details,
        )
